src/rateEngine/billingDeterminants/BlockedTiersInDays.ts

```diff
@@ -14,29 +16,36 @@ class BlockedTiersInDays extends BillingDeterminants {
   private _loadProfile: LoadProfile;
   private _min: Array<number>;
   private _max: Array<number>;
+  private _filters: LoadProfileFilterArgs;
 
   rateElementType = 'Blocked Tier';
   rateClassificationType = 'energy';
   units = 'kWh';
 
-  constructor({ min, max }: BlockedTiersArgs, loadProfile: LoadProfile) {
+  constructor({ min, max, ...filters }: BlockedTiersArgs, loadProfile: LoadProfile) {
     super();
 
     this._loadProfile = loadProfile;
     this._min = min;
     this._max = max;
+    this._filters = filters;
+  }
+
+  filteredLoadProfile(): LoadProfile {
+    return this._loadProfile.filterBy(this._filters);
   }
 
   calculate(): Array<number> {
     const mins = DAYS_PER_MONTH.map((days, i) => days * this._min[i]);
     const maxes = DAYS_PER_MONTH.map((days, i) => days * this._max[i]);
 
-    const expandedLoadProfile = this._loadProfile.expanded();
+    const expandedLoadProfile = this.filteredLoadProfile().expanded();
 
     const monthlyExpandedLoadProfile = Object.values(groupBy(expandedLoadProfile, 'month'));
-    const kwhPerMonth = monthlyExpandedLoadProfile.map((loadProfiles) => sumBy(loadProfiles, 'load'));
+    const kwhByMonth = monthlyExpandedLoadProfile.map((loadProfiles) => sumBy(loadProfiles, 'load'));
 
-    return kwhPerMonth.map((kwh, i) => {
+    return times(12, () => 0).map((zero, i) => {
```

Comment:
The filters cut out all values that don't match, so to keep consistency, we start with a 12 month zero'd array so that we get billing determinants for each month.

Reply:
This is a bit confusing as a glance. Could it be rewritten as:

```js
times(12, (i) => {
   const kwh = kwhByMonth[i] || 0;
   // ...
}
```
?